File: BSharpUnilever/Data/Entities/Product.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace BSharpUnilever.Data.Entities
{
    public class Product
    {
        public int Id { get; set; }

        [Required]
        [MaxLength(255)]
        public string Description { get; set; }

        [MaxLength(255)]
        public string Barcode { get; set; }

        [MaxLength(255)]
        public string SapCode { get; set; }

        [MaxLength(255)]
        public string Type { get; set; } // Home Care (HC), Food Care (FC), Food & Refreshments (F&R), Other (O)

        public bool IsPromo { get; set; }

        public ICollection<SupportRequestLineItem> SupportRequestLineItems { get; set; }
    }
}

File: BSharpUnilever/ClientApp/src/app/data/entities/Product.ts
export class Product {
  Id: number;
  Description: string;
  Barcode: string;
  SapCode: string;
  Type: string;
  IsPromo: boolean;
  IsActive = true;
}

const choicesProductType: { [key: string]: string; } = {
  'HC': 'Home Care',
  'PC': 'Personal Care',
  'F&R': 'Food & Refreshments',
  'O': 'Other',
};

export { choicesProductType };


